### DISCUSSION TITLE
Peak fitting: Wrong height calculation?

### DISCUSSION THREAD

**Discussion by nicocopez** [h2020charisma/ramanchada2#100] [Q&A]
Good morning,
I just realize that sometimes when I perform peak fitting the calculation of the height does not return the right value. If I calculate the height from the returned amplitude it works. So there might be something wrong. 
I do not update the library fo a while thought. Was this issue already detected?

**Comment by vedina** [comment]
depends how do you define height ... it is not always the height from the x axis 

@georgievgeorgi told me sometime ago amplitude and height are defined as in lmfit (see https://lmfit.github.io/lmfit-py/builtin_models.html )  - in fact the amplitude is the area (the integral ) in the fit.

@georgievgeorgi could you confirm / elaborate

**Comment by georgievgeorgi** [comment]
Yes, the peak profiles coming from [lmfit library](https://lmfit.github.io/lmfit-py/index.html) use their definition of amplitude. Is the problem easily reproducible? Does it happen randomly or under a specific conditions like specific peak profile?

**Comment by nicocopez** [comment]
The problem is not with the formula. For some values it works, but at some point it saturates. I am trying to fit the same peak with same noise but different signal intensity, and it saturates when the height reaches 20. Above that does not work anymore. Couple of examples: 
![imagen](https://github.com/h2020charisma/ramanchada2/assets/47475896/221bd768-3689-44d9-98b5-d74e7c5844c6)
I am calculating the height by myself with the formula, so I don't need the thing fixed. However, if you find it useful i can try to code some easy reproducible example.

**Comment by georgievgeorgi** [comment]
The height of the peak is calculated using lambda functions in asteval (an external dependency). If you can give minimal reproducible example it would be easier.
